docs: fix public api tool name

alan-agius4 opened the pull request:
The tool in question is named `TS API Guardian` and not `public API guard`.

alan-agius4 committed:
docs: fix public api tool name

gkalpak reviewed (approved):
While you're at it, it might be helpful to also mention that the "golden files" are kept in the `tools/public_api_guard/` directory.

alan-agius4 committed:
fixup! docs: fix public api tool name

alan-agius4 commented:
@gkalpak PTAL.

gkalpak commented on docs/PUBLIC_API.md:
```diff
@@ -48,9 +48,9 @@ Our peer dependencies (such as TypeScript, Zone.js, or RxJS) are not considered
 ## Golden files
 
-Angular tracks the status of the public API in a *golden file*, maintained with a tool called the *public API guard*.  If you modify any part of a public API in one of the supported public packages, the PR can fail a Circle CI test with failures in `testlogs/tools/public_api_guard`, and an error message that instructs you to accept the golden file.
+Angular tracks the status of the public API in *golden files*. These files are stored in `tools/public_api_guard` and are maintained with a tool called [TS API Guardian](https://www.npmjs.com/package/ts-api-guardian). If you modify any part of a public API in one of the supported public packages, the PR can fail a Circle CI test with failures in `testlogs/tools/public_api_guard`, and an error mes ...
```
```suggestion
Angular tracks the status of the public API in *golden files*. These files are stored in [tools/public_api_guard/](../tools/public_api_guard)` and are maintained with a tool called [TS API Guardian](https://www.npmjs.com/package/ts-api-guardian). If you modify any part of a public API in one of the supported public packages, the PR can fail a Circle CI test with failures in `testlogs/tools/public_api_guard`, and an error message that instructs you to accept the golden file.
```

gkalpak replied:
Wow, I didn't know we publish TS API Guardian 👍

alan-agius4 replied:
Yeah we do :)

alan-agius4 replied:
Change done!

alan-agius4 committed:
fixup! docs: fix public api tool name

gkalpak commented on docs/PUBLIC_API.md:
```diff
@@ -48,9 +48,9 @@ Our peer dependencies (such as TypeScript, Zone.js, or RxJS) are not considered
+Angular tracks the status of the public API in *golden files*. These files are stored in [tools/public_api_guard/](../tools/public_api_guard)` and are maintained with a tool called [TS API Guardian](https://www.npmjs.com/package/ts-api-guardian). If you modify any part of a public API in one of the supported public packages, the PR can fail a Circle CI test with failures in `testlogs/tools/public ...
 
-The public API guard provides a Bazel target that updates the current status of a given package. If you add to or modify the public API in any way, you must use [yarn](https://yarnpkg.com/) to execute the Bazel target in your terminal shell of choice (a recent version of `bash` is recommended).
+The TS API Guardian provides a Bazel target that updates the current status of a given package. If you add to or modify the public API in any way, you must use [yarn](https://yarnpkg.com/) to execute the Bazel target in your terminal shell of choice (a recent version of `bash` is recommended). 
```
😱 😱 😱 

```suggestion
The TS API Guardian provides a Bazel target that updates the current status of a given package. If you add to or modify the public API in any way, you must use [yarn](https://yarnpkg.com/) to execute the Bazel target in your terminal shell of choice (a recent version of `bash` is recommended).
```

gkalpak commented on docs/PUBLIC_API.md:
```diff
@@ -48,9 +48,9 @@ Our peer dependencies (such as TypeScript, Zone.js, or RxJS) are not considered
 ## Golden files
 
-Angular tracks the status of the public API in a *golden file*, maintained with a tool called the *public API guard*.  If you modify any part of a public API in one of the supported public packages, the PR can fail a Circle CI test with failures in `testlogs/tools/public_api_guard`, and an error message that instructs you to accept the golden file.
+Angular tracks the status of the public API in *golden files*. These files are stored in [tools/public_api_guard/](../tools/public_api_guard)` and are maintained with a tool called [TS API Guardian](https://www.npmjs.com/package/ts-api-guardian). If you modify any part of a public API in one of the supported public packages, the PR can fail a Circle CI test with failures in `testlogs/tools/public ...
```
My bad. Forgot to remove the closing `` ` `` in my previous suggestion.

```suggestion
Angular tracks the status of the public API in *golden files*. These files are stored in [tools/public_api_guard/](../tools/public_api_guard) and are maintained with a tool called [TS API Guardian](https://www.npmjs.com/package/ts-api-guardian). If you modify any part of a public API in one of the supported public packages, the PR can fail a Circle CI test with failures in `testlogs/tools/public_api_guard`, and an error message that instructs you to accept the golden file.
```

alan-agius4 committed:
fixup! docs: fix public api tool name

alan-agius4 commented:
@gkalpak PTAL.

IgorMinar commented on docs/PUBLIC_API.md:
```diff
@@ -48,9 +48,9 @@ Our peer dependencies (such as TypeScript, Zone.js, or RxJS) are not considered
 ## Golden files
 
-Angular tracks the status of the public API in a *golden file*, maintained with a tool called the *public API guard*.  If you modify any part of a public API in one of the supported public packages, the PR can fail a Circle CI test with failures in `testlogs/tools/public_api_guard`, and an error message that instructs you to accept the golden file.
+Angular tracks the status of the public API in *golden files*. These files are stored in [tools/public_api_guard/](../tools/public_api_guard) and are maintained with a tool called [TS API Guardian](https://www.npmjs.com/package/ts-api-guardian). If you modify any part of a public API in one of the supported public packages, the PR can fail a Circle CI test with failures in `testlogs/tools/public_ ...
```
```suggestion
Angular tracks the status of the public API in *golden files*. These files are stored in [tools/public_api_guard/](../tools/public_api_guard) and are maintained with a tool called [TS API Guardian](https://www.npmjs.com/package/ts-api-guardian).
If you modify any part of a public API in one of the supported public packages, you must update the golden files that track the public API surface.
If the golden files are not updated correctly, our test suite will detect the problem and the PR will fail to pass the tests.
In such case a test error message will provide instructions for a command to run to update the golden files.
```

alan-agius4 replied:
Updated

IgorMinar commented on docs/PUBLIC_API.md:
```diff
@@ -48,9 +48,9 @@ Our peer dependencies (such as TypeScript, Zone.js, or RxJS) are not considered
+Angular tracks the status of the public API in *golden files*. These files are stored in [tools/public_api_guard/](../tools/public_api_guard) and are maintained with a tool called [TS API Guardian](https://www.npmjs.com/package/ts-api-guardian). If you modify any part of a public API in one of the supported public packages, the PR can fail a Circle CI test with failures in `testlogs/tools/public_ ...
 
-The public API guard provides a Bazel target that updates the current status of a given package. If you add to or modify the public API in any way, you must use [yarn](https://yarnpkg.com/) to execute the Bazel target in your terminal shell of choice (a recent version of `bash` is recommended).
+The TS API Guardian provides a Bazel target that updates the current status of a given package. If you add to or modify the public API in any way, you must use [yarn](https://yarnpkg.com/) to execute the Bazel target in your terminal shell of choice (a recent version of `bash` is recommended).
```
I suggest that you drop this as the error message should give you the right instructions. Or do you think something is missing?

alan-agius4 replied:
The command is already printed in the error message 
https://github.com/angular/angular/blob/1fd4d4f0d4e9ef6d05815e999533e65cff26c51c/tools/ts-api-guardian/lib/cli.ts#L82-L93

alan-agius4 replied:
Though In all fairness that document link shouldn't be there, because not all API guardian users use the same commands. So that document is only valid for Angular/Angular

alan-agius4 replied:
Should we remove it?

IgorMinar replied:
so in that case, we don't need to to add this paragraph to doc, and we should likely drop that link or even better print the link only if the guardian is running in the angular repo (which can be inferred from ENV or even the BAZEL_TARGET itself.

alan-agius4 committed:
fixup! docs: fix public api tool name

alan-agius4 committed:
fixup! docs: fix public api tool name

alan-agius4 committed:
build: disable link to PUBLIC_API docs when running ts-api-guardian in angular repo     

Currently we are showing the following link https://github.com/angular/angular/blob/master/docs/PUBLIC_API.md#golden-files whenever there are updates to the golden files. However this is not correct as this document only applies to the Angular repo as other consumers of ts-api-guardian have different commands, and they store they golden files in different locations.

alan-agius4 committed:
fixup! build: disable link to PUBLIC_API docs when running ts-api-guardian in angular repo

Splaktar commented on docs/PUBLIC_API.md:
```diff
@@ -47,10 +47,10 @@ Our peer dependencies (such as TypeScript, Zone.js, or RxJS) are not considered
+Angular tracks the status of the public API in *golden files*. These files are stored in [tools/public_api_guard/](../tools/public_api_guard) and are maintained with a tool called [TS API Guardian](https://www.npmjs.com/package/ts-api-guardian).
+If you modify any part of a public API in one of the supported public packages, you must update the golden files that track the public API surface.
+If the golden files are not updated correctly, our test suite will detect the problem and the PR will fail to pass the tests.
+In such case a test error message will provide instructions for a command to run to update the golden files.
```
I think that we should leave the note about "a recent version of `bash` is recommended" as I personally ran into that issue and it was a pain to track down without a hint like this. It affects all macOS users as the Apple Bash isn't 'recent' enough.

Splaktar replied:
I'm fine with removing Yarn since it's been added to the error message now. That part was added before the error message was updated.

alan-agius4 replied:
I personally never encountered the issue were I had to update bash. But I am fine with re-adding that part if @IgorMinar & @gkalpak agree.

kapunahelewong commented on docs/PUBLIC_API.md:
```diff
@@ -47,10 +47,10 @@ Our peer dependencies (such as TypeScript, Zone.js, or RxJS) are not considered
+Angular tracks the status of the public API in *golden files*. These files are stored in [tools/public_api_guard/](../tools/public_api_guard) and are maintained with a tool called [TS API Guardian](https://www.npmjs.com/package/ts-api-guardian).
+If you modify any part of a public API in one of the supported public packages, you must update the golden files that track the public API surface.
+If the golden files are not updated correctly, our test suite will detect the problem and the PR will fail to pass the tests.
+In such case a test error message will provide instructions for a command to run to update the golden files.
```
A typo fix and a wording suggestion:
```suggestion
In such cases, a test error message will provide instructions for a command that updates the golden files.
```

alan-agius4 replied:
Done

alan-agius4 committed:
fixup! docs: fix public api tool name

IgorMinar reviewed (approved):
lgtm! thank you!
